code/game/atoms.dm

```diff
@@ -27,7 +27,10 @@
 			var/datum/alternate_appearance/AA = alternate_appearances[aakey]
 			qdel(AA)
 		alternate_appearances = null
-
+	if(viewing_alternate_appearances)
+		for(var/aakey in viewing_alternate_appearances)
+			for(var/aa in viewing_alternate_appearances[aakey])
+				qdel(aa)

```

Comment:
Won't this delete an alternate appearance for *all* viewers if any one of the viewers gets deleted?